src/views/tableButtons.component.test.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import TableButtons from './tableButtons.component';
+import {
+  act,
+  screen,
+  waitForElementToBeRemoved,
+  within,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { renderComponentWithProviders } from '../setupTests';
+import axios from 'axios';
+
+describe('Table buttons', () => {
+  const createView = () => {
+    return renderComponentWithProviders(<TableButtons />);
+  };
+
+  beforeEach(() => {
+    (axios.get as jest.Mock).mockResolvedValue({ data: [] });
+  });
+
+  it('renders the buttons', async () => {
+    let view;
+    await act(async () => {
+      view = createView();
+    });
+    expect(view.asFragment()).toMatchSnapshot();
+  });
+
+  it('opens the filter dialogue when the filters button is clicked and closes when the close button is clicked', async () => {
+    const user = userEvent.setup();
+    createView();
+
+    await user.click(screen.getByRole('button', { name: 'Filters' }));
+
+    const dialogue = await screen.findByRole('dialog', { name: 'Filters' });
+    expect(dialogue).toBeVisible();
+
+    await user.click(within(dialogue).getByRole('button', { name: 'Close' }));
+
+    await waitForElementToBeRemoved(() =>
+      screen.queryByRole('dialog', { name: 'Filters' })
```

Comment:
Should we change this to `expect(screen.queryByRole('dialog', { name: 'Filters' })).not.toBeInTheDocument();`? Otherwise, would this return true if it were still in the document?

Reply:
`waitForElementToBeRemoved` will fail is it's never removed (can be tested by commenting out the previous line clicking the close button) - it will error with `Timed out in waitForElementToBeRemoved.`. It's similar to how you don't need to assert that something exists when using `getBy*` queries - as RTL will fail if they don't exist.